packages/autorest.python/samples/specification/azure-mgmt-test/test/azure-mgmt-test/azure/mgmt/test/aio/_auto_rest_swagger_bat_array_service.py

```diff
@@ -31,14 +33,20 @@ class AutoRestSwaggerBATArrayService:  # pylint: disable=client-accepts-api-vers
     :vartype array: azure.mgmt.test.aio.operations.ArrayOperations
     :param credential: Credential needed for the client to connect to Azure. Required.
     :type credential: ~azure.core.credentials_async.AsyncTokenCredential
-    :param base_url: Service URL. Default value is "http://localhost:3000".
+    :param base_url: Service URL. Default value is None.
     :type base_url: str
     """
 
-    def __init__(
-        self, credential: "AsyncTokenCredential", base_url: str = "http://localhost:3000", **kwargs: Any
-    ) -> None:
-        self._config = AutoRestSwaggerBATArrayServiceConfiguration(credential=credential, **kwargs)
+    def __init__(self, credential: "AsyncTokenCredential", base_url: Optional[str] = None, **kwargs: Any) -> None:
+        _cloud = kwargs.pop("cloud_setting", None) or settings.current.azure_cloud  # type: ignore
```

Comment:
how do we get `http://localhost:3000` in this case? I don't think our azure helpers will give us this value. Should we only do this for ARM generations

Reply:
Yeah, we indeed only do this from ARM generations.
NOTE: this case is [ARM case](https://github.com/Azure/autorest.python/blob/2968c39d808ec9ea538d0ed18ab973240edbdcce/packages/autorest.python/samples/specification/azure-mgmt-test/readme.md?plain=1#L9-L15)